helmclient.go

```diff
@@ -693,12 +720,68 @@ func getTillerImage(pod *corev1.Pod) (string, error) {
 
 	tillerImage := pod.Spec.Containers[0].Image
 	if tillerImage == "" {
-		return "", microerror.Mask(tillerImageNotFoundError)
+		return "", microerror.Mask(tillerImageInvalidError)
 	}
 
 	return tillerImage, nil
 }
 
+func isTillerOutdated(pod *corev1.Pod) error {
+	currentTillerImage, err := getTillerImage(pod)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	currentTillerVersion, err := parseTillerVersion(currentTillerImage)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	desiredTillerVersion, err := parseTillerVersion(tillerImageSpec)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	for i, currentVersion := range currentTillerVersion {
+		desiredVersion := desiredTillerVersion[i]
+		if currentVersion < desiredVersion {
+			return microerror.Maskf(tillerOutdatedError, "%#q older than %#q", currentTillerVersion, desiredTillerVersion)
```

Comment:
Error is only returned if the version is older. As there may be components running different versions of helmclient and we don't want to downgrade the version.

Reply:
Should we still do requests against newer version?

Reply:
My concern is on rolling out the updated `helmclient` version. If we block requests on newer versions too then for example we need to deploy chart-operator and cluster-operator at the same time.

But maybe that is necessary. Not certain yet. I'll keep it in mind and include in my testing.